clash-cores/src/Clash/Cores/Sgmii/Common.hs

```diff
@@ -0,0 +1,115 @@
+module Clash.Cores.Sgmii.Common where
+
+import Clash.Prelude
+
+-- | Format of rxConfReg and txConfReg, size of two data words
+type ConfReg = BitVector 16
+
+-- | Data type that contains a 'BitVector 10' with the corresponding error
+--   condition of the encode function
+data CodeGroup
+  = Cg (BitVector 10)
+  | CgError (BitVector 10)
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Function to check whether a 'CodeGroup' results in a code group
+isCg :: CodeGroup -> Bool
+isCg (Cg _) = True
+isCg _ = False
+
+-- | Function to convert a 'CodeGroup' to a plain 'BitVector 10'
+fromCg :: CodeGroup -> BitVector 10
+fromCg cg = case cg of
+  Cg _cg -> _cg
+  _ -> 0
+
+-- | Data type that contains a 'BitVector 8' with the corresponding error
+--   condition of the decode function
+data DataWord
+  = Dw (BitVector 8)
+  | Cw (BitVector 8)
+  | DwError (BitVector 8)
+  | RdError (BitVector 8)
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Function to check whether a 'DataWord' results in a data word
+isDw :: DataWord -> Bool
+isDw (Dw _) = True
+isDw _ = False
+
+-- | Function to check whether a 'DataWord' results in a data word or
+--   control word
+isValidDw :: DataWord -> Bool
+isValidDw (Cw _) = True
+isValidDw dw = isDw dw
+
+-- | Function to convert a 'DataWord' to a plain 'BitVector 8'
+fromDw :: DataWord -> BitVector 8
+fromDw dw = case dw of
+  Dw _dw -> _dw
+  Cw _dw -> _dw
+  _ -> 0
+
+-- | Defines the possible values for the RUDI output signal of the PCS Receive
+--   block as defined in IEEE 802.3 Clause 36
+data Rudi = C | I | Invalid
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Defines the type of the signal that indicates whether the current received
+--   code group is at an even or odd index in the sequence
+data RxEven = Even | Odd
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Function that makes an Even RxEven Odd, and vice-versa
+nextRxEven :: RxEven -> RxEven
+nextRxEven Even = Odd
+nextRxEven Odd = Even
+
+-- | Defines the type of the signal that indicates whether the transmission is
+--   in sync (Ok) or not (Fail)
+data SyncStatus = Ok | Fail
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Signal that is received by the two PCS blocks from the auto-negotiation
+--   block to indicate the current state of the auto-negotiation block
+data XMit = Configuration | Data | Idle
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Replace a value in a 'BitVector'
+replaceBV :: (KnownNat n) => Index n -> Bit -> BitVector n -> BitVector n
+replaceBV n a bv = pack $ reverse $ replace n a $ reverse $ bv2v bv
+
+-- | Get the value at a given index in a 'BitVector'
+selectBV :: (KnownNat n) => Index n -> BitVector n -> Bit
+selectBV n bv = reverse (bv2v bv) !! n
```

Comment:
[(!)](https://hackage.haskell.org/package/clash-prelude-1.8.1/docs/Clash-Class-BitPack-BitIndex.html)

Reply:
Apparently this was defined but never used, so I just removed it altogether.